pytorch_lightning/core/lightning.py

```diff
@@ -1591,8 +1594,12 @@ def to_torchscript(
                 # if no example inputs are provided, try to see if model has example_input_array set
                 if example_inputs is None:
                     example_inputs = self.example_input_array
+                # dicts are not supported, so show the user an error; not raising an error to show the original error
+                if type(example_inputs) == dict:
```

Comment:
```suggestion
                if isinstance(example_inputs, Mapping):
```

covers more types than `dict`

Reply:
Ah yes, that's better, thank you.